### PR TITLE
All tests

### PR DESCRIPTION
**Pull request by YriyLoz**

### CODE REVIEW

**Comment by TwoySenpai** [comment]
Create .gitignore file and add .idea to this file.
Because you don't need this files on gitHub, by using .gitignore .idea folder will be skipped in commits.

[https://ru.hexlet.io/courses/git_base/lessons/git_gitignore/theory_unit](help link)

File: src/test/java/PetTest.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
+
+
```
Comment (TwoySenpai) [inline src/test/java/PetTest.java]: Remove this spaces from the beginning of the file

File: src/test/java/PetTest.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    }
+
+    @Test(dataProvider = "valueForTests", priority = 3)
+    public void putPetTest(Pet petValue) {
+        httpRequest.body(petValue);
+        Response response = httpRequest.put();
+        boolean result = response.body().asString().contains(new Gson().toJson(petValue));
+        int statusCode = response.getStatusCode();
+        Assert.assertTrue(result);                                                         // Check is it the same object that I have sent
```
Comment (TwoySenpai) [inline src/test/java/PetTest.java]: Move comments before the line which this comment connected with.

Comment (TwoySenpai) [reply src/test/java/PetTest.java]: Also do it for all comments pls.

File: src/test/java/PetTest.java (hunk 1 above)
Comment (TwoySenpai) [reply src/test/java/PetTest.java]: Do it for all files

File: src/test/java/PetTest.java [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    public void postByPetByIdTest(Pet petValue) {
+        RestAssured.baseURI = BASE_URL;
+        RequestSpecification httpRequest = RestAssured.given();
+        httpRequest.contentType("application/x-www-form-urlencoded").formParam("name", petValue.getName()).formParam("status", petValue.getStatus());
+        Response response = httpRequest.post("/" + petValue.getPetId());
+        int statusCode = response.getStatusCode();
+        Assert.assertEquals(statusCode, VALID_STATUS_CODE);
+
+    }
+
```
Comment (TwoySenpai) [inline src/test/java/PetTest.java]: Use 1 line space between blocks
Fix it in all files pls

File: src/test/java/PetTest.java [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+    @Test(dataProvider = "valueForTests", priority = 5)
+    public void postByPetByIdTest(Pet petValue) {
+        RestAssured.baseURI = BASE_URL;
+        RequestSpecification httpRequest = RestAssured.given();
+        httpRequest.contentType("application/x-www-form-urlencoded").formParam("name", petValue.getName()).formParam("status", petValue.getStatus());
+        Response response = httpRequest.post("/" + petValue.getPetId());
+        int statusCode = response.getStatusCode();
+        Assert.assertEquals(statusCode, VALID_STATUS_CODE);
+
```
Comment (TwoySenpai) [inline src/test/java/PetTest.java]: Remove this useless space before close bracket

**Comment by TwoySenpai** [comment]
I think you also have to add binary files to .gitignore bcs u dont need it on github.
This is compiled project
So add target folder to .gitignore

File: src/test/java/PetTest.java [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+    private static final String BASE_URL = "https://petstore.swagger.io/v2/pet";
+    private static RequestSpecification httpRequest;
+
+    @BeforeClass
+    public static void setupForTests() {                                      //Setups for general tests
+        RestAssured.baseURI = BASE_URL;
+        httpRequest = RestAssured.given();
+        httpRequest.accept("application/json");
+        httpRequest.contentType("application/json");
+    }
```
Comment (mburko) [inline src/test/java/PetTest.java]: I noticed that you have the same BeforeClass method in all test classes. How about creating a separate **TestRunner** class where you will init all required for tests data like before/after class and other common data and extend this class in all test classes?

File: src/test/java/PetTest.java [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+
+    @DataProvider
+    public Object[] valueForTests() {
+        Object[] objects = new Object[1];
+        objects[0] = new Pet();
+        return objects;
+    }
```
Comment (mburko) [inline src/test/java/PetTest.java]: How about using this way of DataProvider creation:
```
@DataProvider
    public Object[][] valueForTests() {
        return new Object[][]{{new Pet()}};
    }
```
It looks better for me.

File: src/test/java/PetTest.java [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+
+    @Test(dataProvider = "valueForTests", priority = 5)
+    public void postByPetByIdTest(Pet petValue) {
+        RestAssured.baseURI = BASE_URL;
+        RequestSpecification httpRequest = RestAssured.given();
+        httpRequest.contentType("application/x-www-form-urlencoded").formParam("name", petValue.getName()).formParam("status", petValue.getStatus());
+        Response response = httpRequest.post("/" + petValue.getPetId());
+        int statusCode = response.getStatusCode();
+        Assert.assertEquals(statusCode, VALID_STATUS_CODE);
```
Comment (mburko) [inline src/test/java/PetTest.java]: I think here you also can check that the response body is not null.

File: src/test/java/PetTest.java (hunk 4 above)
Comment (mburko) [inline src/test/java/PetTest.java]: White Space

**Comment by mburko** [comment]
Looks good for a simple API testing.

**Pull request merged**